File: src/main/java/pl/mroz/buddiesapi/domain/account/AccountDomainService.java
package pl.mroz.buddiesapi.domain.account;

import java.util.List;
import java.util.UUID;

public interface AccountDomainService {

    List<Account> getAllAccounts();

    Account getAccount(UUID accountId);

    Account createAccount(Account account);

    Account updateAccount(UUID uuid, Account account);

    void deleteAccount(Account account);
}

File: src/main/kotlin/buddiesapi/domain/account/AccountDomainService.kt
package buddiesapi.domain.account

import java.util.UUID


interface AccountDomainService {
    fun getAllAccounts(): List<Account>

    fun getAccount(accountId: UUID): Account?

    fun createAccount(account: Account): Account

    fun updateAccount(uuid: UUID, account: Account): Account

    fun deleteAccount(account: Account)
}
